packages/browser/src/transports/fetch.ts

```diff
@@ -30,13 +31,17 @@ export function makeFetchTransport(
       ...options.fetchOptions,
     };
 
-    return nativeFetch(options.url, requestOptions).then(response => ({
-      statusCode: response.status,
-      headers: {
-        'x-sentry-rate-limits': response.headers.get('X-Sentry-Rate-Limits'),
-        'retry-after': response.headers.get('Retry-After'),
-      },
-    }));
+    try {
+      return nativeFetch(options.url, requestOptions).then(response => ({
+        statusCode: response.status,
+        headers: {
+          'x-sentry-rate-limits': response.headers.get('X-Sentry-Rate-Limits'),
+          'retry-after': response.headers.get('Retry-After'),
+        },
+      }));
+    } catch (e) {
+      return rejectedSyncPromise(e);
```

Comment:
I am wondering what the long term outcome of this change is... If we go forward with this, we will probably just hide the fact that `fetch` is `undefined` behind a `network error` client report. Is that what we want?

Reply:
Yes this is tricky. I also realized that this change means we don't respect rate-limiting on the SDK at all...